### ISSUE TITLE
Add option "Is large vehicle"

### ISSUE THREAD

**Issue by MacSergey** [CityGecko/CS-AdvancedVehicleOptions#23]
I create a mod that prevents big trucks from delivering goods to small shops. It is possible to add an option "big truck" for transport from the "Delivery" category to this mod? It must set the value to the field VehicleInfo.m_isLargeVehicle. 
I made changes adding an option to the interface. If the author is not opposed to such a change, I would like to commit them
https://steamcommunity.com/sharedfiles/filedetails/?id=2069057130
https://github.com/MacSergey/NoBigTruck

**Comment by CityGecko** [comment]
Hi MacSergey

What is the reason to use VehicleInfo.m_isLargeVehicle (usually reserved for planes/ships) instead of the value of VehicleInfo.m_trailers ? 
With the m_trailers it would be possible to check, if the vehicle has a trailer, which usually would indicate, that this vehicle is a "large vehicle". By this it would be possible to control the spawning of a small box truck instead of a truck/trailer.
AVO does this in [here](https://github.com/CityGecko/CS-AdvancedVehicleOptions/blob/master/AdvancedVehicleOptions/VehicleOptions.cs#L484). I try to understand, what would be more feasible and safe to implement. The best solution would be to break the dependency to AVO, so your mod could run standalone and control settings itself. I like the idea behind your mod, so lets explore for a good implementation.

**Comment by MacSergey** [comment]
To be honest, I started learning how to make mods for CS only a week ago. Therefore, I do not know some features. It seems to me that the flag "m_trailers" is not quite suitable. as there can be a small transport with a trailer and a large truck without a trailer. My mod is ready to work on its own, only it needs a configuration interface. I think your mod is very suitable for this, as it changes the values of other transport flags. The flag "m_isLargeVehicle" seemed to me the most suitable since it is not used in the car and perfectly displays the essence.
Just as an option, my whole mod can be embedded in your mod

**Comment by originalfoo** [comment]
tbh, I think both approaches are needed to some extent, or better still actually measure the vehicle meshes.

there is no guarantee that the large vehicle flag will be set for large vehicles, and very often (in the case of trailers on road vehicles) the front (engine) vehicle is not a large vehicle.

i've been wading through TMPE parking AI recently and saw something in there that might help with measuring vehicles/trailers, I'll see if I can find it again and post below.

those calcs would only need doing once when all the vehicle prefabs/assets are loaded; so even if it adds maybe a couple of seconds to load time, it will not affect in-game performance as the values can be cahced in a dictionary or something (asset name -> length)... we are actually planning something similar for TMPE so we can determine if a vehicle can fit in a parking space more accurately, etc.

**Comment by originalfoo** [comment]
```
vehicleInfo.m_generatedInfo.m_size.x,
vehicleInfo.m_generatedInfo.m_size.z,
```

**Comment by originalfoo** [comment]
btw, at some point in near future, TMPE will have a definition of "what size is a normal vehicle" (again part of the work on revised parking AI). It will be roughly the size of a van, maybe even a small box truck.

Once we have that it will be possible to determine large vehicle roughly like so:

```js
// pseudo code
size = GetVehicleSize(vehicle + trailers);
largeVehicle = size > normalVehicle;
```

**Comment by CityGecko** [comment]
@MacSergey - would this be something you would also look into ? I am still a bit reserved to misuse a value which has different purpose.

@Note to myself, others cordially invited
How to create an own system wide flag so it can be used for deciding delivery with this vehicle to small stores is allowed or not. AVO could manage this, "no big trucks" could read it out.

**Comment by MacSergey** [comment]
@CityGecko  If AVO has a static method that returns a list of large trucks or can tell if the truck is large, then there will be no need to use m_isLargeVehicle flag. Actually exactly the same way you can implement any other functions

**Comment by CityGecko** [comment]
Released with 1.9.3
